js/plugins/vertexai/src/imagen.ts

```diff
@@ -28,72 +28,140 @@ import { GoogleAuth } from 'google-auth-library';
 import { PluginOptions } from './common/types.js';
 import { PredictClient, predictModel } from './predict.js';
 
+/**
+ * See https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/imagen-api.
+ *
+ * TODO: Remove common config schema extension since Imagen models don't support
+ * most of the common config parameters. Also, add more parameters like sampleCount
+ * from the above reference.
+ */
 const ImagenConfigSchema = GenerationCommonConfigSchema.extend({
-  /** Language of the prompt text. */
   language: z
     .enum(['auto', 'en', 'es', 'hi', 'ja', 'ko', 'pt', 'zh-TW', 'zh', 'zh-CN'])
+    .describe('Language of the prompt text.')
+    .optional(),
+  aspectRatio: z
+    .enum(['1:1', '9:16', '16:9', '3:4', '4:3'])
+    .describe('Desired aspect ratio of the output image.')
+    .optional(),
+  negativePrompt: z
+    .string()
+    .describe(
+      'A negative prompt to help generate the images. ' +
+        'For example: "animals" (removes animals), "blurry" ' +
+        '(makes the image clearer), "text" (removes text), or ' +
+        '"cropped" (removes cropped images).'
+    )
+    .optional(),
+  seed: z
+    .number()
+    .int()
+    .min(1)
+    .max(2147483647)
+    .describe(
+      'Any non-negative integer to make output images deterministic. ' +
+        'Providing the same seed number always results in the same output ' +
+        'images.'
+    )
+    .optional(),
+  location: z
+    .string()
+    .describe(
+      'Google Cloud project region. For example: us-central1, europe-west2.'
+    )
     .optional(),
-  /** Desired aspect ratio of output image. */
-  aspectRatio: z.enum(['1:1', '9:16', '16:9', '3:4', '4:3']).optional(),
-  /**
-   * A negative prompt to help generate the images. For example: "animals"
-   * (removes animals), "blurry" (makes the image clearer), "text" (removes
-   * text), or "cropped" (removes cropped images).
-   **/
-  negativePrompt: z.string().optional(),
-  /**
-   * Any non-negative integer you provide to make output images deterministic.
-   * Providing the same seed number always results in the same output images.
-   * Accepted integer values: 1 - 2147483647.
-   **/
-  seed: z.number().optional(),
-  /** Your GCP project's region. e.g.) us-central1, europe-west2, etc. **/
-  location: z.string().optional(),
-  /** Allow generation of people by the model. */
   personGeneration: z
     .enum(['dont_allow', 'allow_adult', 'allow_all'])
+    .describe('Allow generation of people by the model.')
```

Comment:
Its also disallow, or partially allow.

```suggestion
    .describe('Control if/how images of people will be generated by the model.')
```

Reply:
Done.